pkg/backupdriver/backup_driver_controller_base.go

```diff
@@ -339,6 +345,8 @@ func (ctrl *backupDriverController) syncSnapshotByKey(key string) error {
 
 	// Always retrieve up-to-date snapshot CR from API server
 	snapshot, err := ctrl.backupdriverClient.Snapshots(namespace).Get(name, metav1.GetOptions{})
+	// TODO(xyang): Should use snapshotLister for better performance
+	// snapshot, err := ctrl.snapshotLister.Snapshots(namespace).Get(name)
```

Comment:
There is a small window of race condition if we use lister here, since lister read cache for better performance rather than read from API server. 

Suppose when snapshot was just patched to `Snapshotted`, the snapshot lister might possibly still read the stale cache where the same snapshot CR was still in `New`. In this case, we would observe duplicated snapshots taken on the same snapshot CR.

Reply:
Sure, but it also seems that our snapshotting does not support idempotency.  That may be a problem when we implement CSI snapshots.